Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -536,13 +588,90 @@ protected override void ProcessRecord()
             InstancesResource.ImportRequest request = Service.Instances.Import(body, Project, Instance);
             Operation result = request.Execute();
             result = WaitForSqlOperation(result);
-            if (result.Error != null) {
+            if (UploadLocalFile)
+            {
+                bucketService.Objects.Delete(bucketName, fileName).Execute();
+                bucketService.Buckets.Delete(bucketName).Execute();
+            }
+            if (result.Error != null)
+            {
                 foreach (OperationError error in result.Error.Errors)
                 {
+
                     throw new GoogleApiException("Google Cloud SQL API", error.Message + error.Code);
                 }
             }
         }
+
+        private string UploadFile()
```

Comment:
The function needs a doc comment. What does it return? What parameters does it require?

It is generally a good practice to decouple functions from their classes, e.g. if UploadFile could be "static" (not access any class fields or properties) then it is easier to test, debug, etc.

Consider having the function accept a local file path as a parameter -- rather than reading from the class fields -- and then return the newly created GCS object's URI.